### DISCUSSION TITLE
Securing broadcast channel

### DISCUSSION THREAD

**Discussion by RonenEizen** [supabase/supabase#12775] [Questions]
From my understanding anyone who can get a hold of supabase client keys and a channel name can send and receive within the channel. Is there a way to restrict channel access?

**Comment by npgrosser** [comment]
I was wondering the same thing.
The ability to secure broadcast channels would eliminate the need for additional services, like Ably or PubSub, for many use cases. I think a stateless solution would be ideal, so that an access token for a channel can be issued by the application backend.
Ideally with different permissions/claims (e.g. can_read, can_send) 

Or is the channel name already stored and transmitted securely enough to be used as a secret key (e.g. by generating a "hard-to-guess" channel name)?

**Comment by sheiy** [comment]
https://supabase.com/docs/guides/realtime/authorization?queryGroups=language&language=js#helper-functions
